### DISCUSSION TITLE
quiz 147題討論

### DISCUSSION THREAD

**Discussion by Cerosop** [1101-GS4719-C/Python-Community-GS4719-C#1707] [General]
題目如下:
下列關於Python例外處理的敘述，何者正確？
```python
try:
    x = 5 / 0
except ZeroDivisionError:
    print("Division by zero is not allowed.")
finally:
    print("End of program execution.")
```
A、上述程式碼會因為嘗試除以零而產生SyntaxError錯誤。
B、如果遇到ZeroDivisionError錯誤，程式就會立即停止執行。
C、不論是否發生錯誤，最後都會印出 "End of program execution."
D、程式若未發生任何錯誤，則不會印出 "Division by zero is not allowed." 。只有當程式發生某種錯誤時，這句"Division by zero is not allowed."會被印出。 如果一切正常，則不會看到這句話。

答案為C，我能理解C是正確的，但請問有人知道D為甚麼錯了嗎，我看詳解也沒特別說明

**Comment by mrweixd** [comment]
可能問題是在： D、程式若未發生任何錯誤，則不會印出 "Division by zero is not allowed." 。只有當程式發生**某種錯誤**時，這句"Division by zero is not allowed."會被印出。 如果一切正常，則不會看到這句話。
他不能只是某種錯誤，而是ZeroDivisionError才對．
